### DISCUSSION TITLE
Weather widget: add a option to show the Windows 11 widgets menu

### DISCUSSION THREAD

**Discussion by novatiberium** [valinet/ExplorerPatcher#835] [Ideas]
After hearing the recent news of ExplorerPatcher adding the famous Weather widget on the taskbar, that is the best news I've ever heard.

It's unfortunate it hasn't made it to the release builds yet, but work is being done! Although I do have a suggestion - you know how the infamous Cortana button can be programmed to open the Widgets menu instead?

So what if for the weather button in the taskbar, you can have two options: either **show the Google Weather forecast**, or **open the Windows 11 widgets menu,** which will emulate the behavior seen in Windows 10 later versions (since the 11 "widgets" is just the News and Interests but wider), and works a lot better as this was the originally intended purpose.

So having these two options would bring in good functionality to emulate the News and Interests button.

**Comment by valinet** [comment]
Yeah, an issue that I see is the fact that the widgets panel opens from the left while the weather widget is located on the right hand side of the taskbar. The animation would look weird and feel disconnected, and everything would have an 'unfinished' look to it.

Another thing is that Google Weather's provider and Microsoft's provider are different, and it could mean that some locations display different temperatures, which, again, feels a bit odd and disconnected. Here, an alternative would be to pull data from Microsoft instead of Google, but, yeah, it needs additional infrastructure.

Anyway, if someone is interested in doing any work on this, I am open to merging various things/implementations. I personally don't see myself investing much more in this, at least not short term. The thing does its job for the moment, I am looking more into making it a stable experience, polishing any remaining bugs/inconsistencies etc so that we move towards it shipping in a release.

**Reply by valinet** [reply]
Here is a post describing current things that would be nice to be worked on for the weather widget: https://github.com/valinet/ExplorerPatcher/discussions/755

**Comment by XuanChen1234** [comment]
You can ask from @Amrsatrio
